jupyterhub/templates/hub/deployment.yaml

```diff
@@ -110,7 +110,9 @@ spec:
             {{- end }}
           resources:
             {{- .Values.hub.resources | toYaml | trimSuffix "\n" | nindent 12 }}
-          imagePullPolicy: {{ .Values.hub.imagePullPolicy }}
+          {{- with .Values.hub.image.pullPolicy }}
+          imagePullPolicy: {{ . }}
```

Comment:
Rather than explicitly setting this as null, it makes more sense to me to have 

```yaml
{{- if .Values.hub.image.pullPolicy }}
imagePullPolicy: {{ .Values.hub.image.pullPolicy }}
{{- end }}
```

that way the field is actually omitted rather than affirmatively set to null. Does that make sense?

Reply:
Helm's with is also a conditional, it will not render at all if the provided value is falsy. I'm torn between doing it this way and using ifs. Helm's official templates utilizes the `with` way while I just like you got confused about it the first time I saw it.

Reply:
Gotcha. Seems confusing, but always learning. Thanks!